core/integrations/integrations.js

```diff
@@ -0,0 +1,146 @@
+﻿/**
+ * Provides helper functions for handling output integrations.
+ *
+ * Written By:
+ * 		Matthew Knox
+ *
+ * License:
+ *		MIT License. All code unless otherwise specified is
+ *		Copyright (c) Matthew Knox and Contributors 2015.
+ */
+
+var files                   = require.once('../files.js'),
+    fs                      = require('fs'),
+    path                    = require('path'),
+    integrationsLocation    = 'core/core_integrations',
+    cachedIntegrations      = null,
+    selectedIntegrations    = null,
+    started                 = false;
+
+exports.listIntegrations = function () {
+    if (cachedIntegrations) {
+        return cachedIntegrations;
+    }
+    var integ = files.filesInDirectory('./' + integrationsLocation),
+        list = [];
+    for (var i = 0; i < integ.length; i++) {
+        var p = path.join(__dirname, '../../' + integrationsLocation, integ[i]);
+        if (path.extname(integ[i]) !== '') {
+            list.push({
+                name: path.basename(integ[i], path.extname(integ[i])).toLowerCase(),
+                start: p
+            });
+        } else {
+            try {
+                var stat = fs.statSync(p);
+                if (!stat.isDirectory()) {
+                    throw 'Integration must be .js file or directory.';
+                }
+
+                var desc = require(path.join(p, 'package.json'));
+                if (!desc.name || !desc.main) {
+                    throw 'Not enough information provided to describe integration.';
+                }
+
+                list.push({
+                    name: desc.name.replace(/ /g, '-').toLowerCase(),
+                    start: path.join(p, desc.main)
+                });
+            } catch (e) {
+                console.error('Invalid integration installed (syntax error?).');
+                console.critical(e);
+            }
+        }
+    }
+    cachedIntegrations = list;
+    return list;
+};
+
+exports.setIntegrationConfigs = function(platform) {
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        selectedIntegrations[i].instance.platform = platform;
+        selectedIntegrations[i].instance.config = platform.config.loadOutputConfig(selectedIntegrations[i].name);
+
+        for (var name in selectedIntegrations[i].instance.config) {
+            if (name.startsWith("ENV_") && name === name.toUpperCase()) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531155566/issues/source?bid=3282987&fileBranchId=3283570#l65)